src/blenderbim/blenderbim/tool/georeference.py

```diff
@@ -0,0 +1,231 @@
+# BlenderBIM Add-on - OpenBIM Blender Add-on
+# Copyright (C) 2022 Dion Moult <dion@thinkmoult.com>
+#
+# This file is part of BlenderBIM Add-on.
+#
+# BlenderBIM Add-on is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# BlenderBIM Add-on is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with BlenderBIM Add-on.  If not, see <http://www.gnu.org/licenses/>.
+
+
+
+import bpy
+import json
+import blenderbim.core.tool
+import blenderbim.tool as tool
+import ifcopenshell
+import blenderbim.bim.helper
+
+from ifcopenshell.api.unit.data import Data as UnitData
+
+class Georeference(blenderbim.core.tool.Georeference):
+    
+    @classmethod
+    def clear_projected_crs(cls):
+        bpy.context.scene.BIMGeoreferenceProperties.projected_crs.clear()
+    
+    @classmethod
+    def clear_map_conversion(cls):
+        bpy.context.scene.BIMGeoreferenceProperties.map_conversion.clear()
+    
+    @classmethod
+    def set_has_true_north_prop(cls, data):
+        bpy.context.scene.BIMGeoreferenceProperties.has_true_north = bool(data)
+    
+    @classmethod
+    def set_true_north_prop(cls, data):
+        if data:
+            bpy.context.scene.BIMGeoreferenceProperties.true_north_abscissa = str(data[0])
+            bpy.context.scene.BIMGeoreferenceProperties.true_north_ordinate = str(data[1])
+    
+    @classmethod
+    def get_true_north_props(cls, props):
+        true_north = None
+        if props.has_true_north:
+            try:
+                true_north = [float(props.true_north_abscissa), float(props.true_north_ordinate)]
+            except ValueError:
+                self.report({"ERROR"}, "True North Abscissa and Ordinate expect a number") #TODO use the correct report method
```

Comment:
@Moult Here i need help: how can i report the error in a correct way?

Reply:
For now, just turn it into a print() statement to print to console. There are two possible solutions for this I haven't decided which would be better yet.

Option 1: make the core simply return errors and the operator does the reporting, typical client-server approach
Option 2: allow tools to be able to report errors by giving them access to the active operator during dependency injection by the operator

Reply:
Ok, for now i only print the error.

My two cents on this: i think the solution 1 is better because it's more simple; also, i think tool as an instrument to create something so, in case of an error, it reports the error to another component (UI, operator) that can manage the error in its own correct way (print, popup, exception, ...)

Reply:
I agree, please feel free to implement option 1 :)

Reply:
Ok, i'll try to do something about the error report, but i'll do a different PR.